src/RegisterPage.cpp

```diff
@@ -481,6 +482,23 @@ RegisterPage::doUIA(const mtx::user_interactive::Unauthorized &unauthorized)
                 doRegistrationWithAuth(
                   mtx::user_interactive::Auth{session, mtx::user_interactive::auth::Dummy{}});
 
+        } else if (current_stage == mtx::user_interactive::auth_types::registration_token) {
+                auto dialog = new dialogs::TokenRegistration(this);
```

Comment:
Do you think this could be replaced with https://doc.qt.io/qt-5/qinputdialog.html#getText ? That would make the extra dialog redundant, but I'm not sure if you have any plans for the dialog do more that QInputDialog::getText? (Also the label might be slightly different)

Reply:
That's pretty neat :-)